From 81d56e9eb80f7ff96cc7e278e215bc9b4b3c1d31 Mon Sep 17 00:00:00 2001
From: Joey Pabalinas <joeypabalinas@gmail.com>
Date: Mon, 1 Jan 2018 11:24:37 -1000
Subject: [PATCH] arch/x86: add __noreturn __cold to fortify_panic()

Definition of `fortify_panic()` doesn't match the declaration
in include/linux/string.h. Add the missing __noreturn __cold
attributes to `fortify_panic()` and match attribute ordering
in the declaration.

Signed-off-by: Joey Pabalinas <joeypabalinas@gmail.com>
---
 arch/x86/boot/compressed/misc.c | 2 +-
 include/linux/string.h          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/x86/boot/compressed/misc.c b/arch/x86/boot/compressed/misc.c
index 98761a1576ceb5c21b..0e8e8937b34a478433 100644
--- a/arch/x86/boot/compressed/misc.c
+++ b/arch/x86/boot/compressed/misc.c
@@ -428,7 +428,7 @@ asmlinkage __visible void *extract_kernel(void *rmode, memptr heap,
 	return output;
 }
 
-void fortify_panic(const char *name)
+void  __noreturn __cold fortify_panic(const char *name)
 {
 	error("detected buffer overflow");
 }
diff --git a/include/linux/string.h b/include/linux/string.h
index cfd83eb2f926c74622..3adc609a4b6feefe4b 100644
--- a/include/linux/string.h
+++ b/include/linux/string.h
@@ -228,7 +228,7 @@ static inline const char *kbasename(const char *path)
 #define __FORTIFY_INLINE extern __always_inline __attribute__((gnu_inline))
 #define __RENAME(x) __asm__(#x)
 
-void fortify_panic(const char *name) __noreturn __cold;
+void __noreturn __cold fortify_panic(const char *name);
 void __read_overflow(void) __compiletime_error("detected read beyond size of object passed as 1st parameter");
 void __read_overflow2(void) __compiletime_error("detected read beyond size of object passed as 2nd parameter");
 void __read_overflow3(void) __compiletime_error("detected read beyond size of object passed as 3rd parameter");
-- 
2.15.1

